net/wgpeerselector/files/usr/bin/wgpeerselector

```diff
@@ -0,0 +1,505 @@
+	signal.signal(signal.SIGTERM, cleanup("SIGTERM"))
+
+	while true do
+		if connected_peer then
+			state = 'established'
+		else
+			state = 'unconnected'
+		end
+
+		if not time_synchronized then
+			-- Wireguard requires time to be monotonic (always increasing). If
+			-- there was a handshake with a peer once, where we had a higher time
+			-- than our current time, this peer will not accept our handshakes
+			-- until our current time rises above the we had time when the peer had
+			-- the last handshake with us. This usually happens on devices without
+			-- real time clock, like embedded routers. When we rebooted and therefore
+			-- reset our time, the time is lower and handshakes with our last peer
+			-- will fail. Therefore we try to synchronize time by calling ntp.
+			--
+			-- Another possibility, if a boot counter is implemented is outlined here:
+			-- https://lists.zx2c4.com/pipermail/wireguard/2019-February/003850.html
+			time_synchronized = self:sync_ntp()
+		end
+
+		if #self.queued_peers < 1 then
+			self:queue_all_peers()
+		end
+
+		if state == 'unconnected' then
+			local peer = self:queue_pop_random_peer()
+
+			if self:try_connect_to_peer(peer, timeout) then
+				connected_peer = peer
+				log(syslog.LOG_INFO, 'Connection established with '..peer.name..'.')
+			end
+
+		elseif state == 'established' then
+			connected_peer:update_stats_from_kernel()
+
+			if not connected_peer:has_recent_handshake() then
+				connected_peer:uninstall_from_kernel()
+				connected_peer = nil
+				log(syslog.LOG_INFO, 'Connection to '..peer.name..' lost.')
+			else
+				-- check connections every 5 seconds
+				sleep(5)
+			end
+		else
+			log(syslog.LOG_CRIT, 'unknown state')
+			os.exit(1)
+		end
+
+		if #self.queued_peers < 1 then
+			-- this prevents the process from escalating if e.g. installing peers
+			-- constantly fails.
+			sleep(5)
+		end
+
+		if self:wait_for_iface() then
+			-- We needed to wait here, so interface was gone inbetween. Most likely
+			-- the interface has been reset, so there is none of our peers installed.
+			-- However, to make sure, we call the cleanup routine.
+			self:cleanup()
+			connected_peer = nil
+		end
+	end
+end
+
+-- INIT
+
+-- Parse command arguments
+_G.is_verbose = false
+local actions = {}
+local skip = false
+local change_group_to = nil
+local iface = "wg0"
+
+for i = 1, #arg do
+	if skip then
+		skip = false
+	elseif arg[i] == "--interface" or arg[i] == "-i" then
+		iface = arg[i+1]
+		skip = true
+	elseif arg[i] == "--pid-file" or arg[i] == "-p" then
+		actions['pid'] = arg[i+1]
+		skip = true
+	elseif arg[i] == "--group" then
+		change_group_to = arg[i+1]
+		skip = true
+	elseif arg[i] == "--version" then
+		actions['version'] = true
+	elseif arg[i] == "--verbose" or arg[i] == "-v" then
+		is_verbose = true
+	elseif arg[i] == "--help" or arg[i] == "-h" then
+		actions['help'] = true
+	end
+end
+
+if actions['version'] then
+	print("wgpeerselector v1")
+	os.exit(0)
+end
```

Comment:
Ususally Gluon packages are not versioned, so I'd drop that.

Reply:
This doesn't really make sense to me. Versioning software is always a good idea, I would say. Furthermore it seems like they actually have. See e.g. [here](https://github.com/freifunk-gluon/packages/blob/master/net/respondd/Makefile#L4).

Reply:
All Gluon packages are versioned as v1, as otherwise they are not built with theOpenWrt buildsystem.

So I would either version the package implicitly using the commit hash of the packages repo or drop this version command completely. Versioning of installed packages can be tracked using opkg with the first approach.